src/a-runtime-console/kubernetes/model/kubernetesresource.model.ts

```diff
@@ -61,12 +61,47 @@ export class KubernetesResource implements BaseEntity {
 
     this.icon = this.annotations['fabric8.io/iconUrl'] || templateAnnotations['fabric8.io/iconUrl'] || this.defaultIconUrl();
 
+    // lets fix up old image URIs
+    if (this.isFabric8Icon(this.icon, "camel")) {
+      this.icon = "/_assets/images/icon-stack-camel.png";
+    } else if (this.isFabric8Icon(this.icon, "funktion")) {
+      this.icon = "/_assets/images/icon-stack-funktion.png";
+    } else if (this.isFabric8Icon(this.icon, "go")) {
+      this.icon = "/_assets/images/icon-stack-go.png";
+    } else if (this.isFabric8Icon(this.icon, "java")) {
+      this.icon = "/_assets/images/icon-stack-java.png";
+    } else if (this.isFabric8Icon(this.icon, "maven")) {
+      this.icon = "/_assets/images/icon-stack-maven.png";
+    } else if (this.isFabric8Icon(this.icon, "microservice")) {
+      this.icon = "/_assets/images/icon-stack-microservice.png";
+    } else if (this.isFabric8Icon(this.icon, "nodejs")) {
+      this.icon = "/_assets/images/icon-stack-nodejs.png";
+    } else if (this.isFabric8Icon(this.icon, "python")) {
+      this.icon = "/_assets/images/icon-stack-python.png";
+    } else if (this.isFabric8Icon(this.icon, "rails")) {
+      this.icon = "/_assets/images/icon-stack-rails.png";
+    } else if (this.isFabric8Icon(this.icon, "spring")) {
+      this.icon = "/_assets/images/icon-stack-spring.png";
+    } else if (this.isFabric8Icon(this.icon, "swift")) {
+      this.icon = "/_assets/images/icon-stack-swift.png";
+    } else if (this.isFabric8Icon(this.icon, "vertx")) {
+      this.icon = "/_assets/images/icon-stack-vertx.png";
+    } else if (this.isFabric8Icon(this.icon, "wildfly")) {
+      this.icon = "/_assets/images/icon-stack-wildfly.png";
+    }
     // TODO any other annotations we should look for?
     this.description = this.annotations['description'] || templateAnnotations['description'] || '';
 
     this.openShiftConsoleUrl = openShiftBrowseResourceUrl(this, currentOAuthConfig());
   }
 
+  isFabric8Icon(url, name) {
+    if (url && (url.startsWith("img/icons/") || url.startsWith("/img/icons/"))) {
```

Comment:
I thought the idea was to point to the `/assets/images` directory?

Reply:
that is the idea. 

This code is about transforming **old** fabric8 icon URLs generated via the fabric8-maven-plugin and annotated onto released `Deployment` resources to URLs which work nicely on the OSIO console. i.e. the code replaces `img/icons/spring.png -> /_assets/images/icon-stack-spring.png`